amp/WEB-INF/src/org/dgfoundation/amp/onepager/components/features/items/AmpMEIndicatorFeaturePanel.java

```diff
@@ -69,36 +71,66 @@ public AmpMEIndicatorFeaturePanel(String id, String fmName, final IModel<Indicat
+        final AmpIndicatorGlobalValue globalTargetVal = new AmpIndicatorGlobalValue(AmpIndicatorGlobalValue.TARGET);
 
         final Model<Boolean> valuesSet = new Model<Boolean>(false);
 
-        for (AmpIndicatorValue val : values.getObject()){
+        for (AmpIndicatorGlobalValue val : indicator.getObject().getIndicatorValues()){
 
-            switch (val.getValueType()) {
+            switch (val.getType()) {
                 case AmpIndicatorValue.BASE:
-                    val.copyValuesTo(baseVal);
+                    val.copyValuesTo(globalBaseVal);
                     break;
                 case AmpIndicatorValue.TARGET:
-                    val.copyValuesTo(targetVal);
+                    val.copyValuesTo(globalTargetVal);
                     valuesSet.setObject(true);
                     break;
                 default:
                     break;
             }
         }
 
-        final Label indicatorBaseValueLabel = new Label("base", new PropertyModel<>(baseVal, "value"));
+        final Label indicatorBaseValueLabel = new Label("base", new LoadableDetachableModel<String>() {
+            @Override
+            protected String load() {
+                return globalBaseVal.getOriginalValue() != null ? String.valueOf(globalBaseVal.getOriginalValue()) : "N/A";
+            }
+        });
         add(indicatorBaseValueLabel);
 
-        final Label indicatorBaseDateLabel = new Label("baseDate", new PropertyModel<>(baseVal, "valueDate"));
+        final Label indicatorBaseDateLabel = new Label("baseDate", new LoadableDetachableModel<String>() {
+            @Override
+            protected String load() {
+                if (globalBaseVal.getOriginalValueDate() != null) {
+                    SimpleDateFormat format = new SimpleDateFormat("M/dd/yy hh:mm a");
+                    return format.format(globalBaseVal.getOriginalValueDate());
+                } else {
+                    return "N/A";
+                }
+            }
+        });
         add(indicatorBaseDateLabel);
 
-        final Label indicatorTargetValueLabel = new Label("target", new PropertyModel<>(targetVal, "value"));
+        final Label indicatorTargetValueLabel = new Label("target", new LoadableDetachableModel<String>() {
+            @Override
+            protected String load() {
+                return globalTargetVal.getOriginalValue() != null ? String.valueOf(globalTargetVal.getOriginalValue()) : "N/A";
+            }
+        });
         add(indicatorTargetValueLabel);
 
-        final Label indicatorTargetDateLabel = new Label("targetDate", new PropertyModel<>(targetVal, "valueDate"));
+        final Label indicatorTargetDateLabel = new Label("targetDate", new LoadableDetachableModel<String>() {
+            @Override
+            protected String load() {
+                if (globalTargetVal.getOriginalValueDate() != null) {
+                    SimpleDateFormat format = new SimpleDateFormat("M/dd/yy hh:mm a");
```

Comment:
For the comment above
```suggestion
                    SimpleDateFormat dateFormat = new SimpleDateFormat(Objects.requireNonNull(FeaturesUtil.getGlobalSettingValue(GlobalSettingsConstants.DEFAULT_DATE_FORMAT)) + "h:mm a");
```

Reply:
Done